firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -223,12 +223,20 @@ private final void doRegistration() {
       PersistedInstallationEntry persistedInstallationEntry =
           persistedInstallation.readPersistedInstallationEntryValue();
 
+      // Default value of iidToken
```

Comment:
you did not change that in this CL, but this method is another example for a try-catch block that covers the entire method.
That smells to me that this method should throw that exception.
Otherwise, can we maybe move the try-catch block around the part that actually can cause the exception?

Reply:
I need to pass the exceptions to the caller of the SDK API asynchronously. Hence the need to have a try-catch block covering the entire method.

It doesn't let me highlight the block that hasn't changed. I can explain you what I mean by this in-person.